### DISCUSSION TITLE
Simplifying setup environment variables

### DISCUSSION THREAD

**Discussion by domdomegg** [mathesar-foundation/mathesar#3347] [Feature Ideas]
I think it should be easy as possible to get Mathesar up and running in an appropriate configuration. This would help gain new users.

One of the steps on the critical path to onboarding is having to set envrionment variables. Currently there are four required environment variables: https://docs.mathesar.org/configuration/env-variables/. Each of these is an opportunity to make a mistake, and adds complexity to the setup process.

I think we can narrow it down to just the DJANGO_DATABASE_URL:

- ALLOWED_HOSTS to default to `*`, so it can be removed from required configuration. This may require a review of the code to check where we might return the host header or make decisions on it, to ensure this does not leave the default configuration to [Host header attacks](https://docs.djangoproject.com/en/4.2/topics/security/#host-headers-virtual-hosting) - however, it's reasonably likely that this is possible to do.
- MATHESAR_DATABASES to be removed and replaced by the UI connection manager (i.e. see #3223, #3344)
- SECRET_KEY to be automatically generated on startup, and then stored in the database

**Comment by seancolsen** [comment]
@domdomegg thanks for suggesting this. Because your suggestion is a code-related change, I've converted it into an [issue](https://github.com/mathesar-foundation/mathesar/issues/3355). That will help us track it better with other proposals for code-related changes. We're trying to use GitHub discussions for more open-ended Q&A going forward, though admittedly we still have lots of previous discussions which don't fit this pattern. Anyway, I'm closing this discussion now to track it in the issue.
